java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -590,16 +589,19 @@ public final ColumnVector subVector(int start, int end) {
    *
    * Note that this is very similar to the output from a PartitionedTable.
    *
-   * @param indices the indexes to split with
-   * @return A new ColumnVector array with slices from the original ColumnVector
+   * NOTE: Nothing is copied out from the vector and the slices will only be relevant for the
+   * lifecycle of the underlying ColumnVector
+   *
+   * @param indices the indices to split with
+   * @return A new ColumnView array with slices from the original ColumnView
    */
-  public final ColumnVector[] split(int... indices) {
+  public ColumnView[] split(int... indices) {
```

Comment:
I'm not sure changing the semantics of an existing method is the best approach.  Given the vastly different semantics on ownership between the two aliasing `split` methods, I'd rather see a new method here, like `splitAsViews`.  That also means a `ColumnView` can do a split that copies to `ColumnVector` as it did before rather than forcing the user to either wastefully upgrade a view into a `ColumnVector` or manually copy the resulting views to get their desired behavior.